bok_choy/a11y/a11y_audit.py

```diff
@@ -0,0 +1,267 @@
+        format are expected to be unmodified.
+        """
+        if path:
+            self.rules_file = os.path.abspath(path)
+
+    @abstractmethod
+    def set_rules(self, rules, ignore=False, include_type="rule"):
+        """
+        Overrides the default list of rules to be run.
+
+        Raises:
+
+            `NotImplementedError` if this isn't overwritten in the ruleset
+                specific implementation.
+        """
+        raise NotImplementedError(
+            "The ability to specify rules has not been implemented."
+        )
+
+    @abstractmethod
+    def set_scope(self, include=None, exclude=None):
+        """
+        Overrides the default scope (part of the DOM) to inspect.
+
+        Raises:
+
+            `NotImplementedError` if this isn't overwritten in the ruleset
+                specific implementation.
+        """
+        raise NotImplementedError(
+            "The ability to specify scope has not been implemented."
+        )
+
+
+class A11yAudit(object):
+    """
+    Allows auditing of a page for accessibility issues.
+
+    The ruleset to use can be specified by the environment variable
+    `BOKCHOY_A11Y_RULESET`. Currently, there are two ruleset implemented:
+
+    `axe_core`:
+
+        * Ruleset class: AxeCoreAudit
+        * Ruleset config: AxeCoreAuditConfig
+        * This is default ruleset.
+
+    `google_axs`:
+
+        * Ruleset class: AxsAudit
+        * Ruleset config: AxsAuditConfig
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, browser, url, config=None, *args, **kwargs):
+        """
+        Sets ruleset to be used.
+
+        Args:
+            browser: A phantomjs browser instance
+            url: URL of the page to test
+            config: (optional) A11yAuditConfig or subclass of A11yAuditConfig
+        """
+        super(A11yAudit, self).__init__(*args, **kwargs)
+        self.url = url
+        self.browser = browser
+        self.config = config or self.default_config
+
+    def _phantomjs_setup(self):
+        """
+        Checks that phantomjs is being used.
+
+        Returns: (ghostdriver_url, sessions) where sessions is a list of
+            session ids.
+
+        Raises: `NotImplementedError` if not using phantomjs.
+        """
+        if self.browser.name != 'phantomjs':
+            msg = (
+                'Accessibility auditing is only supported with PhantomJS as'
+                ' the browser.'
+            )
+            raise NotImplementedError(msg)
+
+        # The ghostdriver URL will be something like this:
+        # 'http://localhost:33225/wd/hub'
+        ghostdriver_url = self.browser.service.service_url
+
+        # Get the session_id from ghostdriver so that we can inject JS into
+        # the page.
+        resp = requests.get('{}/sessions'.format(ghostdriver_url))
+        sessions = resp.json()
+
+        return ghostdriver_url, sessions
+
+    def _verify_file_exists(self):
```

Comment:
This should probably be named _verify_rules_file_exists, as it doesn't take a filename as a parameter and it is specifically checking self.rules_file's existence.